PowerEditor/src/ScintillaComponent/Buffer.cpp

```diff
@@ -1137,6 +1137,29 @@ SavingStatus FileManager::saveBuffer(BufferID id, const TCHAR * filename, bool i
 		}
 	}
 
+	const wchar_t* filePath = buffer->getFullPathName();
+	wchar_t dir[MAX_PATH];
+	wcscpy_s(dir, MAX_PATH, filePath);
+	::PathRemoveFileSpecW(dir);
+	ULARGE_INTEGER freeBytesForUser;
+	 
+	BOOL getFreeSpaceRes = ::GetDiskFreeSpaceExW(dir, &freeBytesForUser, nullptr, nullptr);
+	if (getFreeSpaceRes != FALSE)
+	{
+		int64_t fileSize = buffer->getFileLength();

```

Comment:
If we are saving a new copy of the same file under different name then the original file will not be overwritten and its size should not be added to the free space calculation, right? That is if a new file copy is made this way...

Reply:
Do you mean "Save As..." or "Save a Copy As..."? If you do, then what you worry about won't happen, because the "filename" argument will be a new name so the size won't be added to the free space calculation.

OTOH, your remark makes me test with the both commands mentioned above to "Save as an existing file". It is rejected because this PR doesn't consider other existing file as the target. So in case of source & destination are not the same file,  and if source file length is smaller than destination one, it could be saved but operation is not permitted.

I do not want to manage the above case, for the reason that: 1. it'll make the code complex 2. it'll be more safe while no room left.